### PR TITLE
P2: new signup flow

### PR DESCRIPTION
**Pull request by annemirasol**
#### Changes proposed in this Pull Request

* Partially implements the new signup flow as described in p9lV3a-2Lp-p2 :
    * Defines a new P2 flow (`p2-new`) that adds a new step and  puts user signup/login before site creation
    * Note: the old P2 flow (`p2`) will not _yet_ be removed, and will still be the main flow, until more steps are implemented for the new flow.

#### Testing instructions
`yarn install && ENABLE_FEATURES=no-force-sympathy yarn start` or Calypso live

1. Regression test: make sure that the old flow, `/start/p2`, still works as expected. Go to `calypso.localhost:3000/start/p2` and go through the signup flow. 
2. Testing the new flow:
    - Note: `/start/p2-new` is not the complete and final version of the revised flow described in p9lV3a-2Lp-p2. It will not yet be linked to by the P2 marketing page. However, it is a complete flow, i.e. it allows users to log in or sign up, and create P2 workspaces.
    - To test, go to `calypso.localhost:3000/start/p2-new` and go through the signup flow.

#### Screens
Screen Shot 2021-11-29 at 7 23 15 PM

### CODE REVIEW

File: client/signup/main.jsx [hunk 1]
```diff
@@ -667,9 +664,16 @@ class Signup extends Component {
...
+		const classes = [ 'signup__step' ];
+		if ( isP2Flow( this.props.flowName ) ) {
+			classes.push( `is-${ this.props.stepName }` );
+		} else {
+			classes.push( `is-${ kebabCase( this.props.stepName ) }` );
+		}
+
 		return (
 			<div className="signup__step" key={ stepKey }>
-				<div className={ `signup__step is-${ kebabCase( this.props.stepName ) }` }>
```
Comment (lamosty) [inline client/signup/main.jsx]: Why do we need to make an exception for the P2 signup flow here and below?

File: client/signup/steps/p2-get-started/index.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import { Button } from '@wordpress/components';
```
Comment (lamosty) [inline client/signup/steps/p2-get-started/index.jsx:4]: Could we split these into internal, external and WP dependencies, please?

File: client/signup/steps/p2-get-started/style.scss [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	background-color: #000000;
+
+	.p2-step-wrapper__header-logo {
+		svg path {
+			fill: var( --p2-color-white );
+		}
+	}
+
+	.p2-get-started__option {
+		border-bottom: 1px solid #808080;
```
Comment (lamosty) [inline client/signup/steps/p2-get-started/style.scss]: Could we store these colors as variables, please?

File: client/signup/p2-step-wrapper/style-p2v2.scss [hunk 4]
```diff
@@ -0,0 +1,158 @@
+@import url( 'https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600;900&display=swap' );
+@import '@wordpress/base-styles/breakpoints';
+@import '@wordpress/base-styles/mixins';
+@import 'calypso/assets/stylesheets/_p2-extends.scss';
+
+.is-p2v2 {
```
Comment (lamosty) [inline client/signup/p2-step-wrapper/style-p2v2.scss]: I'm not sure if we could somehow optimize these class names so that we don't have to put everything under `.is-p2v2`. Or maybe it's okay? I guess since we are reusing so much (ie p2-step-wrapper) but want custom styling for the v2.

File: client/signup/p2-step-wrapper/style-p2v2.scss [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+		@include break-medium {
+			padding: 64px 20px 0;
+		}
+
+		@include break-huge {
+			padding: 80px 20px 0;
+		}
+
+		input {
```
Comment (lamosty) [inline client/signup/p2-step-wrapper/style-p2v2.scss]: Nit: could we name all the elements and target them by class names, please?

**Review by lamosty — COMMENTED** [review]
Great work Anne!

I'm not sure what's the original UX but I'd say we should make the whole area clickable, not just the ">" arrow (e.g. "Create a new workspace" and the text below) on the p2-get-started step.

Also, I suppose it's expected at this point but clicking on both "create a new workspace" and "join an existing workspace" takes me to to the same place.

**Comment by evilluendas** [comment]
Aside from visual tweaks, I've noticed that if you create a wpcom user but cancel the flow before setting up an actual site, you receive this email:

image

It'd be awesome to disable it if folks are going through the P2 flow, if it's possible.

**Comment by david-gonzalez-a8c** [comment]
Thanks Anne, this looks great!

I'd like to make some tweaks:

1. Font sizes for the headings (like "Create a new workspace") and the paragraphs below (like "Start a new P2...") are a point higher. They are 20px/16px and in Figma we have 19px/15px
2. I don't know how to check that out, but is the background color #00101C? Seems darker to me
3. The separator lines should be #FFF at 30% transparency. If we want to use a solid color we can use #4D5860
4. I'd put the link to the next step over the whole block and not just the chevron icon. I'm attaching a gif for an animation I think we can use on hover. Please note that for it to work better I added an extra 10px padding to the left and right of each block.
5. In mobile, the logo and first title get a little too close. I added a 1.5em margin top to the title and seems to work fine.
6. I'd check the distances between the separator lines and the title/paragraph as they seem higher than 24px, which is what we have in Figma.

Summing up the changes in a screenshot for clarity. 

Thanks!

Signup tweaks

signup gif

**Comment by evilluendas** [comment]
We've also noticed some changes to the `Sign up` screen that may be caused by the new dark selection screen, so I wanted to bring it up in case it's something that need to be fixed here:

- The "Already have a WordPress.com account?" link is white, so it's invisible. In the current /start/p2 flow we also show a different message, "Already have a WordPress.com account? Log in instead" which would be nice to keep.
- When you reduce the screen to mobile sizes (I think?), a strange Recaptcha message appears.
- The "Continue with Google" button doesn't work without allowing third party cookies in Chrome, which may be related to the previous thing. The current `/start/p2` flow works perfectly fine so I guess it's an issue with Calypso.live, since it's using a different domain? The popover explaining how to enable the cookies covers the "Continue with email" primary button and that is pretty annoying, but I suspect we won't have to worry about this once it's in production under WordPress.com.

Here's a recording of all these issues, in case it helps. I suspect the only real thing to keep in mind is the "Log in instead" thing:

https://user-images.githubusercontent.com/820374/143300936-28314f1e-2779-4c18-af05-ef5360edc9e2.mov

Please disregard all of this if it's supposed to be fixed in a different step 🙏

**Comment by annemirasol** [comment]
Thank you everyone for the quick reviews! 🙇 I will be working on all the feedback little by little, and will re-ping once everything's done.

**Comment by annemirasol** [comment]
> 
> It'd be awesome to disable it if folks are going through the P2 flow, if it's possible.

Done!  D70639-code should skip sending this email to P2 users

File: client/signup/main.jsx (hunk 1 above)
Comment (annemirasol) [reply client/signup/main.jsx]: Mmyeah not too sold on doing this, but `kebabCase` was making the classname `is-p-2`/`is-p-2-v-2` (instead of `is-p2`/`is-p2v2`), which I found confusing at first. But what do you think? Is it worth the extra logic just for a "proper-looking" classname?

File: client/signup/p2-step-wrapper/style-p2v2.scss (hunk 4 above)
Comment (annemirasol) [reply client/signup/p2-step-wrapper/style-p2v2.scss]: Correct, it's just to ensure I'm not affecting any of the current styles. The `.is-p2v2` wrapper is temporary and will be removed entirely when the revised flow becomes the main flow. `style-p2v2.scss` will become `style.scss` as `p2v2` becomes `p2`.

File: client/signup/main.jsx (hunk 1 above)
Comment (lamosty) [reply client/signup/main.jsx]: Ah, I see. I'm thinking whether it makes sense to call the flow `p2v2`. Maybe we could do something like `p2-new`? I suppose anything that would satisfy the original `kebabCase` so we don't have to refactor it. 😅  Since we'll be changing the name to `p2` after it's done anyways and I'd say `p2v2` is not user friendly url part anyways.

**Comment by annemirasol** [comment]
@david-gonzalez-a8c What is the hover background color in the gif you attached?

**Comment by david-gonzalez-a8c** [comment]
It's #FFFFFF at 3% opacity.

Alternatively, we can use #061925 as a solid color

**Comment by annemirasol** [comment]
TO DO:
- [x] Fix font sizes (row title and description)
- [x] Fix background color (should be `#00101c`)
- [x] Fix separator color (should be `#4d5860`)
- [x] Make entire row clickable, with hover styles
- [x] Add top margin to page title
- [x]  Fix row margins/paddings
- [x] Disable 'Finish creating your site' marketing email (D70639)
- [x] Fix invisible Sign Up screen login link
- [x] Remove Recaptcha stuff appearing in mobile vps
- [x] Rename flow to `p2-new`
- [x] Remove kebabCase exceptions
- [x] Use variables for CSS colors
- [x] Fix non-specific selectors (removed) 
- [x] Tag internal, external dependencies

File: client/signup/steps/p2-get-started/style.scss (hunk 3 above)
Comment (annemirasol) [reply client/signup/steps/p2-get-started/style.scss]: Done!

File: client/signup/steps/p2-get-started/index.jsx (hunk 2 above)
Comment (annemirasol) [reply client/signup/steps/p2-get-started/index.jsx:4]: Done!

File: client/signup/p2-step-wrapper/style-p2v2.scss (hunk 5 above)
Comment (annemirasol) [reply client/signup/p2-step-wrapper/style-p2v2.scss]: These rules were actually redundant and have been removed.

**Comment by annemirasol** [comment]
> The "Continue with Google" button doesn't work without allowing third party cookies in Chrome, which may be related to the previous thing. The current /start/p2 flow works perfectly fine so I guess it's an issue with Calypso.live, since it's using a different domain? The popover explaining how to enable the cookies covers the "Continue with email" primary button and that is pretty annoying, but I suspect we won't have to worry about this once it's in production under WordPress.com.

Correct, this one is only a problem when testing in local or calypso.live.

File: client/signup/main.jsx (hunk 1 above)
Comment (annemirasol) [reply client/signup/main.jsx]: Changed the flowname to `p2-new` (that was its original name! 😅). Some of the classnames became, e.g. `is-p-2-new`, `is-p-2-get-started`, but I guess that's an acceptable tradeoff compared to having extra classname logic for P2 flows.

**Comment by annemirasol** [comment]
I think I've addressed all feedback. This one's ready for another look, thank you!

File: client/assets/stylesheets/_p2-vars.scss [hunk 6]
```diff
@@ -16,6 +16,10 @@
 	--p2-color-button: #f3f3f3;
 	--p2-color-button-dark: #eaeaea;
 
+	--p2-color-background-dark: #00101c;
```
Comment (lamosty) [inline client/assets/stylesheets/_p2-vars.scss:19]: Looks like we already have the color defined in this file above as `--p2-color-text`.

**Review by lamosty — APPROVED** [review]
Thanks for addressing all the feedback!

I've tested both the new and the current flows and they all seem to work! I haven't tested WP.com signup flows but we haven't changed anything related to them so we should be fine.

File: client/assets/stylesheets/_p2-vars.scss (hunk 6 above)
Comment (annemirasol) [reply client/assets/stylesheets/_p2-vars.scss:19]: Yup! I thought I'd keep it separate though, as we may be tweaking the background color separate from the text color.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7004573

Thank you @annemirasol for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7004573).
